nimare/meta/cbma/base.py

```diff
@@ -260,30 +247,63 @@ def _p_to_summarystat(self, p, null_method=None):
         if null_method is None:
             null_method = self.null_method
 
-        if null_method.startswith("analytic"):
+        if null_method.startswith("approximate"):
             assert "histogram_bins" in self.null_distributions_.keys()
-            assert "histogram_weights" in self.null_distributions_.keys()
+            assert "histweights_corr-none_method-approximate" in self.null_distributions_.keys()
 
-            hist_weights = self.null_distributions_["histogram_weights"]
+            hist_weights = self.null_distributions_["histweights_corr-none_method-approximate"]
             # Desired bin is the first one _before_ the target p-value (for consistency
-            # with the empirical null).
+            # with the montecarlo null).
             ss_idx = np.maximum(0, np.where(hist_weights <= p)[0][0] - 1)
             ss = self.null_distributions_["histogram_bins"][ss_idx]
 
-        elif null_method == "empirical":
-            assert "empirical_null" in self.null_distributions_.keys()
-            null_dist = np.sort(self.null_distributions_["empirical_null"])
+        elif null_method == "montecarlo":
+            assert "histogram_bins" in self.null_distributions_.keys()
+            assert "histweights_corr-none_method-montecarlo" in self.null_distributions_.keys()
+            null_dist = np.sort(
+                self.null_distributions_["histweights_corr-none_method-montecarlo"]
+            )
             n_vals = len(null_dist)
             ss_idx = np.floor(p * n_vals).astype(int)
             ss = null_dist[-ss_idx]
         else:
-            raise ValueError("Argument 'null_method' must be one of: 'analytic', 'empirical'.")
+            raise ValueError("Argument 'null_method' must be one of: 'approximate', 'montecarlo'.")
 
         return ss
 
-    def _run_fwe_permutation(self, params):
+    def _run_montecarlo_permutation(self, params):
```

Comment:
This is pretty darn slow, but the histogram is probably necessary, as noted by @tyarkoni in #389.